src/query/graphite/native/aggregation_functions.go

```diff
@@ -89,7 +90,33 @@ func maxSeries(ctx *common.Context, series multiplePathSpecs) (ts.SeriesList, er
 	return combineSeries(ctx, series, wrapPathExpr("maxSeries", ts.SeriesList(series)), ts.Max)
 }
 
-// divideSeries divides one series list by another series
+func divideSeriesHelper(ctx *common.Context, dividendSeries, divisorSeries ts.Series, metadata block.ResultMetadata) (*ts.Series, error) {
+	normalized, minBegin, _, lcmMillisPerStep, err := common.Normalize(ctx, ts.SeriesList{
+		Values:   []*ts.Series{&dividendSeries, &divisorSeries},
+		Metadata: metadata,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// NB(bl): Normalized must give back exactly two series of the same length.
+	dividend, divisor := normalized.Values[0], normalized.Values[1]
+	numSteps := dividend.Len()
+	vals := ts.NewValues(ctx, lcmMillisPerStep, numSteps)
+	for i := 0; i < numSteps; i++ {
+		dividendVal := dividend.ValueAt(i)
+		divisorVal := divisor.ValueAt(i)
+		if !math.IsNaN(dividendVal) && !math.IsNaN(divisorVal) && divisorVal != 0 {
+			value := dividendVal / divisorVal
+			vals.SetValueAt(i, value)
+		}
+	}
+	name := fmt.Sprintf("divideSeries(%s,%s)", dividend.Name(), divisor.Name())
```

Comment:
Need to make the function name come from the caller? Should be `divideSeries` for divide series and `divideSeriesLists` for divide series list.

Reply:
Believe it or not, if you take a look at the `graphite` source code, you will see that the individual series will be named `divideSeries(X, X)`, even if it is generated by `divideSeriesLists`

Take a look here --> https://github.com/graphite-project/graphite-web/blob/17a34e7966f7a46eded30c2362765c74eea899cb/webapp/graphite/render/functions.py#L901

Reply:
Hah, ok that's interesting.

Sounds good - can you add a comment to describe why this is done for context for future readers?

Reply:
done!